control/trajectory_follower_nodes/src/lateral_controller_node.cpp

```diff
@@ -0,0 +1,521 @@
+// Copyright 2021 The Autoware Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "trajectory_follower_nodes/lateral_controller_node.hpp"
+
+#include "tf2_ros/create_timer_ros.h"
+
+#include <algorithm>
+#include <deque>
+#include <limits>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace autoware
+{
+namespace motion
+{
+namespace control
+{
+namespace trajectory_follower_nodes
+{
+namespace
+{
+using namespace std::chrono_literals;
+
+template <typename T>
+void update_param(
+  const std::vector<rclcpp::Parameter> & parameters, const std::string & name, T & value)
+{
+  auto it = std::find_if(
+    parameters.cbegin(), parameters.cend(),
+    [&name](const rclcpp::Parameter & parameter) { return parameter.get_name() == name; });
+  if (it != parameters.cend()) {
+    value = static_cast<T>(it->template get_value<T>());
+  }
+}
+}  // namespace
+
+LateralController::LateralController(const rclcpp::NodeOptions & node_options)
+: Node("lateral_controller", node_options)
```

Comment:
Could you create sample launch file?

Reply:
these are a packages porting from .auto so changes applied to them should be as minimal as possible.
https://github.com/autowarefoundation/autoware.universe/issues/117